distribution/pull_v2.go

```diff
@@ -88,6 +92,11 @@ func (p *v2Puller) Pull(ctx context.Context, ref reference.Named) (err error) {
 func (p *v2Puller) pullV2Repository(ctx context.Context, ref reference.Named) (err error) {
 	var layersDownloaded bool
 	if !reference.IsNameOnly(ref) {
+		var err error
+		ref, err = p.checkTrusted(ctx, ref)
+		if err != nil {
+			return allowV1Fallback(err)
+		}
```

Comment:
alright, v1 fallback is now disabled if there's any error checking the image's signature